index.bs

```diff
@@ -1889,6 +1935,141 @@ The [=remote end steps=] with |session| and <var ignore>command parameters</var>
 
 </div>
 
+#### The browser.createUserContext Command #### {#command-browser-createUserContext}
+
+The <dfn export for=commands>browser.createUserContext</dfn> command creates a [=user context=].
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      browser.CreateUserContext = (
+        method: "browser.createUserContext",
+        params: EmptyParams,
+      )
+      </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+      browser.createUserContextResult = {
+        userContext: browser.UserContext
+      }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for browser.createUserContext">
+
+The [=remote end steps=] are:
+
+1. Let |user context| be the result of performing implementation defined steps to create an [=empty user context=].
+
+1. Let |result| be a [=/map=] matching the
+   <code>browser.createUserContextResult</code> production with the <code>userContext</code>
+   field set to |user context|'s [=user context id=].
+
+1. Return [=success=] with data |result|.
+
+</div>
+
+#### The browser.closeUserContext Command #### {#command-browser-closeUserContext}
+
+The <dfn export for=commands>browser.closeUserContext</dfn> command closes a user context and all browsing contexts in it without running <code>beforeunload</code> handlers.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      browser.CloseUserContext = (
+        method: "browser.closeUserContext",
+        params: {
+          userContext: browser.UserContext
+        },
+      )
+      </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+      <pre class="cddl">
+      EmptyResult
+      </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for browser.closeUserContext">
+
+The [=remote end steps=] with |command parameters| are:
+
+1. Let |user context| be |command  parameters|["<code>userContext</code>"].
+
+1. If the [=user context=] with the [=user context id=] equal to |user context|
+   does not exist in the user agent, return [=error=] with [=error code=] [=invalid
+   argument=]
+
+1. [=Close=] any [=top-level browsing contexts=] that belong to the |user
+   context| without [=prompting to unload=].
+
+    Note: This implicitly only affects browsing contexts that were under
+    automation in the first place.
+
+    Issue: Should we allow closing the default context? Probably not.
+
+1. Perform implementation defined steps to remove the |user context| from the
```

Comment:
I think this would be easier if we just said that an implementation has a set of active user contexts, which we add to on creation, or remove from on close, rather than throwing in "implementation-defined" everywhere.